publication-commons/src/main/java/no/unit/nva/doi/models/Doi.java

```diff
@@ -0,0 +1,77 @@
+package no.unit.nva.doi.models;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import nva.commons.utils.JacocoGenerated;
+
+/**
+ * Doi class for working with Dois.
+ *
+ * <p>Use {@link Doi#builder()} for constructing a new Doi instance.
+ */
+public abstract class Doi {
+
+    public static final URI DOI_PROXY = URI.create("https://doi.org/");
+    public static final String ERROR_PROXY_URI_MUST_BE_A_VALID_URL = "Proxy URI must be a valid URL";
+    private static final String FORWARD_SLASH = "/";
+
+    public static ImmutableDoi.Builder builder() {
+        return ImmutableDoi.builder();
+    }
+
```

Comment:
Do we want copyOf here?

```
public static Doi copyOf(Doi doi) {
  return immutableDoi.copyOf(doi)
}
```

Reply:
Same as I added `Doi#builder` , easier and looks nicer. Then you end up using:

```
Doi.copyOf(..)
or 
Doi.builder()

instead of having to type the ImmutableDoi name, like ImmutableDoi.builder()
``

Reply:
It hides the `implementation` behind the Doi class.

Reply:
We removed `copyOf` as we dont need that functionality.